src/main/java/com/alipay/oceanbase/rpc/table/ObTable.java

```diff
@@ -394,8 +391,28 @@ public ObPayload execute(final ObPayload request) throws RemotingException,
         } catch (Exception ex) {
             throw new ObTableConnectionStatusException("check status failed", ex);
         }
-
-        return realClient.invokeSync(connection, request, obTableExecuteTimeout);
+        boolean needReconnect = false;
+        int retryTimes = 0;
+        ObPayload payload = null;
+        do {
+            retryTimes++;
+            try {
+                if (needReconnect) {
+                    String msg = String.format("Receive error: tenant not in server and reconnect it, ip:{}, port:{}, tenant id:{}, retryTimes: {}",
+                            connection.getObTable().getIp(), connection.getObTable().getPort(), connection.getTenantId(), retryTimes);
+                    connection.reConnectAndLogin(msg);
+                    needReconnect = false;
+                }
+                payload = realClient.invokeSync(connection, request, obTableExecuteTimeout);
+            } catch (ObTableException ex) {
+                if (ex instanceof ObTableTenantNotInServerException && retryTimes < 2) {
+                    needReconnect = true;
+                } else {
+                    throw ex;
+                }
+            }
+        } while (needReconnect && retryTimes < 2);
```

Comment:
Make sure we only retry one time.

Reply:
Yet, now we only retry one time and if not success, it will throw this Exception to the upper function